src/Traits/HasNameTrait.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace WikibaseSolutions\CypherDSL\Traits;
+
+use function bin2hex;
+use function ceil;
+use function openssl_random_pseudo_bytes;
+use function substr;
+
+trait HasNameTrait
+{
+    use ErrorTrait;
+
+    private string $name;
+
+    private static int $automaticVariableLength = 32;
+
+    /**
+     * Returns the name.
+     *
+     * @return string
+     */
+    public function getName(): string
+    {
+        if (!isset($this->name)) {
+            $this->configureName(null, 'var');
+        }
+
+        return $this->name;
+    }
+
+    /**
+     * Generates a unique random identifier.
+     *
+     * @note It is not entirely guaranteed that this function gives a truly unique identifier. However, because the
+     * number of possible IDs is so huge, it should not be a problem.
+     *
+     * @param string $prefix The prefix to put before the name. Must start with a letter to adhere to cypher namings.
+     *
+     * @param int|null $length The length of the generated name in bytes.
+     *
+     * @return string
+     */
+    private function generateName(string $prefix = 'var', int $length = null): string
+    {
+        $length ??= self::$automaticVariableLength;
+
+        return $prefix . substr(bin2hex(openssl_random_pseudo_bytes(ceil($length / 2))), 0, $length);
```

Comment:
I think `openssl_random_pseudo_bytes` might be a bit overkill here; it is somewhat expensive, the names generated here need not be cryptographically secure, and using it here might decrease entropy when some other function (that actually needs cryptographically secure numbers) does a subsequent call to `openssl_random_pseudo_bytes` or a related function.

But this is really a nitpick.

Reply:
I completely agree. It was already there in the library's existing implementation, so I moved it. I changed it to work with random_bytes now

Reply:
`random_bytes` still tries to be cryptographically secure. Either `rand` or `mt_rand` are relatively cheap pseudo-random number generators that should be used for this purpose.

Reply:
I did not know that! It appears there no simple built-in function in that case. So I combined mt_rand with dechex to solve the problem. What do you think?

Reply:
great!